### PR TITLE
feat(Modelarts): add a datasource to get model templates

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
Running tool: /usr/local/go/bin/go test -timeout 9000000000s -run ^TestAccDatasourceModelTemplates_basic$ github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/modelarts -v

=== RUN   TestAccDatasourceModelTemplates_basic
=== PAUSE TestAccDatasourceModelTemplates_basic
=== CONT  TestAccDatasourceModelTemplates_basic
--- PASS: TestAccDatasourceModelTemplates_basic (6.37s)
PASS
ok  	github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/modelarts	6.406s

```

### CODE REVIEW

File: docs/data-sources/modelarts_model_templates.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String) Specifies the region in which to query the data source.
+  If omitted, the provider-level region will be used.
+
+* `type` - (Optional, String) The type of model. The valid value are **Classification** and **Common**.
```
Comment (deer-hang) [inline docs/data-sources/modelarts_model_templates.md]:
```suggestion
* `type` - (Optional, String) The type of model. The valid values are **Classification** and **Common**.  
```

Please check the other parts.

File: docs/data-sources/modelarts_model_templates.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+The following arguments are supported:
+
+* `region` - (Optional, String) Specifies the region in which to query the data source.
+  If omitted, the provider-level region will be used.
+
+* `type` - (Optional, String) The type of model. The valid value are **Classification** and **Common**.  
+
+* `engine` - (Optional, String) The AI engine.  
+  The valid value are **Caffe**, **Caffe1.0 CPU**, **Caffe1.0 GPU**, **MXNet**, **MXNet1.2.1**,
+   **MindSpore**, **PyTorch**, **PyTorch1.0**, **TensorFlow**, and **TensorFlow1.8**.
```
Comment (deer-hang) [inline docs/data-sources/modelarts_model_templates.md:28]:
```suggestion
   **MindSpore**, **PyTorch**, **PyTorch1.0**, **TensorFlow** and **TensorFlow1.8**.
```

Please check the other parts.

File: docs/data-sources/modelarts_model_templates.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+* `environment` - (Optional, String) Model runtime environment.  
+  The valid value are **ascend-arm-py2.7**, **python2.7**, and **python3.6**.
+
+* `keyword` - (Optional, String) Keywords to search in name and description. Fuzzy match is supported.
```
Comment (deer-hang) [inline docs/data-sources/modelarts_model_templates.md]:
```suggestion
* `keyword` - (Optional, String) Keywords to search in name or description. Fuzzy match is supported.
```

File: huaweicloud/services/modelarts/data_source_huaweicloud_modelarts_model_templates.go [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+	return &sc
+}
+
+func modelTemplatesInputsSchema() *schema.Resource {
+	sc := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The id of the input parameter.`,
```
Comment (deer-hang) [inline huaweicloud/services/modelarts/data_source_huaweicloud_modelarts_model_templates.go]:
```suggestion
				Description: `The ID of the input parameter.`,
```

File: docs/data-sources/modelarts_model_templates.md (hunk 1 above)
Comment (hmbig2) [reply docs/data-sources/modelarts_model_templates.md]: done

File: docs/data-sources/modelarts_model_templates.md (hunk 3 above)
Comment (hmbig2) [reply docs/data-sources/modelarts_model_templates.md]: ok

File: huaweicloud/services/modelarts/data_source_huaweicloud_modelarts_model_templates.go (hunk 4 above)
Comment (hmbig2) [reply huaweicloud/services/modelarts/data_source_huaweicloud_modelarts_model_templates.go]: ok

File: docs/data-sources/modelarts_model_templates.md (hunk 2 above)
Comment (hmbig2) [reply docs/data-sources/modelarts_model_templates.md:28]: comma(,) before `and`, it's correct

**Pull request merged**